src/couch_query_servers.erl

```diff
@@ -169,6 +178,19 @@ sum_values(Value, Acc) when is_number(Value), is_list(Acc) ->
     sum_arrays(Acc, [Value]);
 sum_values(Value, Acc) when is_list(Value), is_number(Acc) ->
     sum_arrays([Acc], Value);
+sum_values({Props}, Acc) ->
+    case lists:keyfind(<<"error">>, 1, Props) of
```

Comment:
Also unfortunately but we can't return anything but an EJSON value from this because there's no way to insert a convert-to-EJSON step later on. And any attempt to do so would be outside of the standard reduce function APIs and would prevent any non-builtin reduce from participating in that protocol (cause other language can only pass JSON around).